framework/src/problems/FEProblemBase.C

```diff
@@ -785,6 +787,12 @@ FEProblemBase::initialSetup()
 {
   TIME_SECTION("initialSetup", 2, "Performing Initial Setup");
 
+  if (_set_nonlinear_convergence_name)
+    getConvergence(_nonlinear_convergence_name).initialSetup();
+  else
+    Moose::out << "No convergence criteria specified. Using default convergence criteria."
+               << std::endl;
```

Comment:
this is confusing, the user did pass convergence criteria parameters, just not a convergence object